lib/queue.js

```diff
@@ -670,6 +675,7 @@ Queue.prototype.processJob = function(job){
       return;
     }
 
+    clearTimer();
```

Comment:
I think it'd be slightly more readable to make this a part of the promise chain, eg

```
return clearTimer().
    then(function() {
       return job.moveToCompleted(data);
    })
```